test/core/linter_test.py

```diff
@@ -585,3 +589,21 @@ def test_delayed_exception():
     de = runner.DelayedException(ve)
     with pytest.raises(ValueError):
         de.reraise()
+
+
+def test__attempt_to_change_templater_warning(caplog):
+    """Test warning if user tries to change templater in .sqlfluff file in subdirectory."""
+    initial_config = FluffConfig(configs={"core": {"templater": "jinja"}})
+    lntr = Linter(config=initial_config)
+    updated_config = FluffConfig(configs={"core": {"templater": "dbt"}})
+    logger = logging.getLogger("sqlfluff")
+    original_propagate_value = logger.propagate
+    try:
```

Comment:
Worked around the logging issue by temporarily setting the propagate value to `True` after reading[ this](https://stackoverflow.com/questions/59875983/why-is-caplog-text-empty-even-though-the-function-im-testing-is-logging) SO post.

I was concerned that playing with the logging values like this would give false positive tests but I also manually tested that this warning appears when it is supposed to and it does.

what do you think?
@alanmcruickshank, @barrywhart, @NiallRees

Reply:
I was able to get this test to pass on commits before your temporary propagate switch when running with the whole suite by setting https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/cli/commands.py#L67 to True. The CLI tests run before `linter_test.py` which must explain why this test passes in isolation but not when running the entire suite.

I'm happy with your solution here, though I wonder whether it makes sense to just set `fluff_logger.propagate = True` in the CLI module when testing so we don't run into this again. What do you think @alanmcruickshank ?

Reply:
btw there's more reading on the "propagate" issue: https://github.com/pytest-dev/pytest/issues/3697

seems like there isn't a universal way to deal with it :(

Reply:
I think I'm just going to merge this as we know what the issue is, we can revisit it later if needed. I'll make an issue to keep track.

Reply:
@NiallRees, @alanmcruickshank, @dmateusp, @barrywhart Thanks for the support with this issue. It was my first open source contribution and a positive experience (although more work than I expected haha).

Reply:
Congratulations! You picked a tough first contribution 😄 - nice work.